src/internal/kopia/snapshot_manager.go

```diff
@@ -0,0 +1,232 @@
+	"github.com/kopia/kopia/snapshot"
+	"github.com/pkg/errors"
+
+	"github.com/alcionai/corso/src/pkg/logger"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+const (
+	// Kopia does not do comparisons properly for empty tags right now so add some
+	// placeholder value to them.
+	defaultTagValue = "0"
+	// Kopia CLI prefixes all user tags with "tag:"[1]. Maintaining this will
+	// ensure we don't accidentally take reserved tags and that tags can be
+	// displayed with kopia CLI.
+	// (permalinks)
+	// [1] https://github.com/kopia/kopia/blob/05e729a7858a6e86cb48ba29fb53cb6045efce2b/cli/command_snapshot_create.go#L169
+	userTagPrefix = "tag:"
+)
+
+type snapshotManager interface {
+	FindManifests(
+		ctx context.Context,
+		tags map[string]string,
+	) ([]*manifest.EntryMetadata, error)
+	LoadSnapshots(ctx context.Context, ids []manifest.ID) ([]*snapshot.Manifest, error)
+}
+
+type ownersCats struct {
+	resourceOwners map[string]struct{}
+	serviceCats    map[string]struct{}
+}
+
+func serviceCatTag(p path.Path) string {
+	return p.Service().String() + p.Category().String()
+}
+
+func makeTagPair(k string) (string, string) {
+	return userTagPrefix + k, defaultTagValue
+}
+
+// tagsFromStrings returns a map[string]string with the union of both maps
+// passed in. Currently uses placeholder values for each tag because there can be
+// multiple instances of resource owners and categories in a single snapshot.
+func tagsFromStrings(oc *ownersCats) map[string]string {
+	res := make(map[string]string, len(oc.serviceCats)+len(oc.resourceOwners))
+
+	for k := range oc.serviceCats {
+		tk, tv := makeTagPair(k)
+		res[tk] = tv
+	}
+
+	for k := range oc.resourceOwners {
+		tk, tv := makeTagPair(k)
+		res[tk] = tv
+	}
+
+	return res
+}
+
+// getLastIdx searches for manifests contained in both foundMans and metas
+// and returns the most recent complete manifest index. If no complete manifest
+// is in both lists returns -1.
+func getLastIdx(
+	foundMans map[manifest.ID]*snapshot.Manifest,
+	metas []*manifest.EntryMetadata,
+) int {
+	// Minor optimization: the current code seems to return the entries from
+	// earliest timestamp to latest (this is undocumented). Sort in the same
+	// fashion so that we don't incur a bunch of swaps.
+	sort.Slice(metas, func(i, j int) bool {
+		return metas[i].ModTime.Before(metas[j].ModTime)
+	})
+
+	// Search newest to oldest.
+	for i := len(metas) - 1; i >= 0; i-- {
+		m := foundMans[metas[i].ID]
+		if m == nil || len(m.IncompleteReason) > 0 {
+			continue
+		}
+
+		return i
+	}
+
+	return -1
+}
+
+// findRecentManifests searches through mans and returns the most recent
+// complete manifest (if one exists) and maybe the most recent incomplete
+// manifest. If the newest incomplete manifest is more recent than the newest
+// complete manifest then adds it to the returned list. Otherwise no incomplete
+// manifest is returned. Returns nil if there are no complete or complete
+// manifests in mans.
+func findRecentManifests(
```

Comment:
Nit: the comment does a good job explaining the behavior.  Still, the function name itself is easily misinterpreted.
```suggestion
func manifestsSinceLastComplete(
```